script/urdfToBlender.py

```diff
@@ -17,6 +17,19 @@
 from bpy.types import Operator
 
 
+def createGeometricShape(iDynTree_solidshape):
+    if iDynTree_solidshape.isSphere():
+        bpy.ops.mesh.primitive_uv_sphere_add(radius=iDynTree_solidshape.asSphere().getRadius())
+    elif iDynTree_solidshape.isCylinder():
+        cylinder = iDynTree_solidshape.asCylinder()
+        bpy.ops.mesh.primitive_cylinder_add(radius=cylinder.getRadius(), depth=cylinder.getLength())
+    elif iDynTree_solidshape.isBox():
+        bpy.ops.mesh.primitive_cube_add(size=iDynTree_solidshape.asBox().getX()) # Seems that blender support only cubes
```

Comment:
Probably you can then deform a cube to obtain a box? Anyhow, I would add a check on getX / getY / getZ to be equal, and if they are not print an error. Silently failing and producing a cube instead seems to be a source of hard to debug problems in hte future otherwise.

Reply:
You are right, the only problem is that since they are floating-point numbers the comparison may lead to some false positives unless I don't define a tolerance in the comparison

>Probably you can then deform a cube to obtain a box?

Eh, good question, looking at [this](https://docs.blender.org/manual/en/latest/modeling/meshes/primitives.html) there is not the concept of the box in the primitives (but of monkey mask yes :D). I am sure that deforming it should make the trick, but I have to investigate about it.

Reply:
I added the check https://github.com/robotology/blender-robotics-utils/commit/33e469b7bd9fe02a15bb09f17f6de8e428cd1e4d but I will see how to support also parallelepiped